src/utils/filter.expenses.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+const filterByQuery = (params, items) => {
+  const { userId, from, to, categories } = params;
+  let newExpenses = [...items];
+
+  if (userId) {
+    newExpenses = newExpenses.filter(exp => exp.userId === +userId);
+  }
+
+  if (categories) {
+    newExpenses = newExpenses.filter(exp => exp.category === categories);
+  }
+
+  if (from && to) {
+    newExpenses = newExpenses
+      .filter(exp => exp.spentAt > from && exp.spentAt < to);
+  }
```

Comment:
please use condition in one `.filter` method, because now it will invoke filter a lot of times of there is a multiple filters in query

Reply:
Thanks, done.